src/devices/src/virtio/vsock/packet.rs

```diff
@@ -222,8 +222,14 @@ impl VsockPacket {
 
     /// Writes the local copy of the packet header to the guest memory.
     pub fn commit_hdr(&self, mem: &GuestMemoryMmap) -> Result<()> {
-        mem.write_obj(self.hdr, self.hdr_addr)
-            .map_err(VsockError::GuestMemoryMmap)
+        if self.len() < u32::MAX - VSOCK_PKT_HDR_SIZE as u32 {
```

Comment:
On a second thought, isn't this already checked [here](https://github.com/firecracker-microvm/firecracker/blob/afe9c0623f12156b22c814e86f7b4420906c71c9/src/devices/src/virtio/vsock/packet.rs#L180) ?

Reply:
This is inside `from_tx_virtq_head` function, it is not part of our code path.

Reply:
That is nice, did not know there is a MAX_PKT_BUF_SIZE defined for this. So seeing this I suggest to create an auxiliary function in Vsockpacket with [this check](https://github.com/firecracker-microvm/firecracker/blob/afe9c0623f12156b22c814e86f7b4420906c71c9/src/devices/src/virtio/vsock/packet.rs#L180) and then use it from there. We can call it from `commit_hdr` before trying to `write_obj`.

Reply:
Done